helm/cloud-provider-azure/templates/_image.tpl

```diff
@@ -1,24 +1,42 @@
 {{/*
+Determine the default image repository based on Kubernetes version.
+*/}}
+{{- define "image.repo" -}}
+    {{- $root := (index . 0) }}
+    {{- $repoOverride := (index . 1) }}
+    {{- if $repoOverride -}}
+        {{ $repoOverride }}
+    {{- else if semverCompare "< 1.34.0-0" $root.Capabilities.KubeVersion.Version -}}
```

Comment:
@feiskyer Is the plan to publish new patch releases of older minor versions to the new repo too? e.g. v1.30.14 under `/oss/kubernetes` and v1.30.15 under `/oss/v2/kubernetes`. Or will v1.34.0+ be the only versions under `/oss/v2/kubernetes`?